src/app/user/modules/records/components/record-form/record-form.component.html

```diff
@@ -127,17 +127,17 @@ <h4 class="card-title">培训记录填报</h4>
             <div formArrayName="files">
               <h3>
                 <span i18n>附件</span>
-                <button mat-button type="button" color="primary" (click)="addFile()" [disabled]="files.length > 2"
+                <button mat-button type="button" color="primary" (click)="addFile()" [disabled]="files.length > 2-originalAttachments.length"
                   i18n>添加新附件</button>
               </h3>
               <div *ngIf="hasOriginalAttachments">
+              <p class="text-danger" *ngIf="files.length > 2-originalAttachments.length">您最多能上传3个附件。</p>
```

Comment:
If you move the warning here, then user won't be able to see this warning when creating a new record, cuz `hasOriginalAttachments` is `False`.